Sources/Datadog/Core/Upload/DataUploadWorker.swift

```diff
@@ -90,3 +91,22 @@ internal class DataUploadWorker: DataUploadWorkerType {
         }
     }
 }
+
+extension DataUploadConditions.Blocker: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case let .battery(level: level, state: state):
+            return "🔋 Battery state is: \(state) (\(level)%)"
+        case .lowPowerModeOn:
+            return "🔌 Low Power Mode is: enabled"
+        case let .networkReachability(description: description):
+            return "📡 Network reachability is: " + description
+        }
+    }
+}
+
+extension Array where Element == DataUploadConditions.Blocker {
+    var description: String {
+        "❌ [upload was skipped because: " + self.map { $0.description }.joined(separator: " AND ") + "]"
+    }
```

Comment:
i think it was 💯 with
```swift
var description: String {
  if self.isEmpty { return "✅" }
  else { return "❌ [upload..." }
}
```
i was wondering why you removed ✅ case?

Reply:
I did not remove the `isEmpty` case, it wasn't there, but that's a great idea :)